### ISSUE TITLE
Investigate loss spikes

### ISSUE THREAD

**Issue by dlwh** [stanford-crfm/mistral#115]
We occasionally get large spikes in the training loss during training. For example:
![image](https://user-images.githubusercontent.com/9633/157739392-560ac44d-c6cf-4393-86b7-fc9879f9cee8.png)

The system ends up recovering, but it can take quite a long time. This may not be worth worrying about because the system does recover, but we should understand this.

One hypothesis is we get "bad batches" where the data is garbage, is thus high perplexity, and gives us bad gradients.

Another is instability related to fp16.
